### DISCUSSION TITLE
"Generated SQL" does not show SQL when query has an error

### DISCUSSION THREAD

**Discussion by adam-nielsen** [grafana/grafana#38181] [Dashboards, Panels & Visualizations]
If I run an SQL query and click the "Generated SQL" link, then all the placeholders are replaced with their actual values and in the sidebar I can see the raw SQL sent to the server, which is very useful for debugging.

However if the server is returning an error, then Grafana does not show the raw SQL so it is very difficult to diagnose the error, particularly when the error appears to be in the way one of the placeholders is formatted.

This is what "Generated SQL" looks like when there is an error, no query can be seen:

![screenshot](https://user-images.githubusercontent.com/278772/100397171-07422300-3094-11eb-9662-1914a63258e0.png)

**How to reproduce it (as minimally and precisely as possible)**:

Put in an invalid SQL statement and try to view the generated SQL for it.

**Environment**:
- Grafana version: 7.3.4
- Data source type & version: PostgresSQL + TimescaleDB
- OS Grafana is installed on: DockerHub grafana/grafana:latest
- User OS & Browser: Firefox + Chrome

**Comment by aocenas** [comment]
For future reference this is how it looks like with successful query:
![Screenshot from 2020-11-27 12-16-35](https://user-images.githubusercontent.com/1014802/100443685-788cd080-30aa-11eb-96ae-8de261c88633.png)

I assume the full interpolated query should be shown even if server returns error (probably only if the query building fails we would have nothing to show).

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by SharkWipf** [comment]
This "feature request" (or more accurately, bug), is still around at least when using Postgres/TimescaleDB as backend.
This makes it quite complicated to figure out what, exactly, Grafana's macros work out to.
The queries of course still get executed, and if you check the database logs you can see the full query, but this is often not possible or complicated.

**Comment by gabor** [comment]
in the current version of Grafana, i recommend opening the `Query Inspector`, and in it's `Query` tab you can see the exact query that was sent to the database. this works also when there was an error.

for example, here is a screenshot that happens when there is a syntax error in the query:
<img width="585" alt="sql" src="https://github.com/grafana/grafana/assets/51989/5edf968e-2ba6-4c38-a49e-4c9a9192bed6">

i hope this solves the problem.

if it does not solve the problem, please open a new github issue, and describe why the current solution is not sufficient. thanks!
